Sources/Implementation/DefaultDatafileHandler.swift

```diff
@@ -110,27 +110,38 @@ class DefaultDatafileHandler: OPTDatafileHandler {
             guard let request = self.getRequest(sdkKey: sdkKey) else { return }
             
             let task = session.downloadTask(with: request) { (url, response, error) in
-                var result = OptimizelyResult<Data?>.failure(.datafileDownloadFailed("Failed to parse"))
+                var result = OptimizelyResult<Data?>.failure(.datafileLoadingFailed(sdkKey))
+
+                let returnCached = { () in
```

Comment:
we would not get a notification change because it would be the same revision as the one in memory.  in memory and in cache would be equal on a network failure.

Reply:
In Android, we decide for datafileUpdated by watching the cache datafile, but in Swift, we decide it based on when DatafileHandler returns a valid datafile. It should return .success(nil) for reading from cache, then the caller knows it's from cache.

Reply:
i updated the updater so that if the revision is the same, do nothing.